src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,380 @@
+        /// </summary>
+        /// <param name="diagnosticId"> The ID of the diagnostic.</param>
+        /// <param name="targetNode"> The node to be cast.</param>
+        /// <param name="targetNodeType"> Output the type of "targetNode".</param>
+        /// <param name="potentialConversionTypes"> Output the potential conversions types that "targetNode" can be cast to</param>
+        /// <returns>
+        /// True, if the target node has at least one potential conversion type, and they are assigned to "potentialConversionTypes"
+        /// False, if the target node has no conversion type.
+        /// </returns>
+        private static bool TryGetTargetTypeInfo(
+            SemanticModel semanticModel, SyntaxNode root, string diagnosticId, ExpressionSyntax targetNode,
+            CancellationToken cancellationToken, [NotNullWhen(true)] out ITypeSymbol? targetNodeType,
+            out ImmutableArray<ITypeSymbol> potentialConversionTypes)
+        {
+            potentialConversionTypes = ImmutableArray<ITypeSymbol>.Empty;
+
+            var targetNodeInfo = semanticModel.GetTypeInfo(targetNode, cancellationToken);
+            targetNodeType = targetNodeInfo.Type;
+
+            if (targetNodeType == null)
+                return false;
+
+            // The error happens either on an assignement operation or on an invocation expression.
+            // If the error happens on assignment operation, "ConvertedType" is different from the current "Type"
+            using var _ = ArrayBuilder<ITypeSymbol>.GetInstance(out var mutablePotentialConversionTypes);
+            if (diagnosticId == CS0266
+                && targetNodeInfo.ConvertedType != null
+                && !targetNodeType.Equals(targetNodeInfo.ConvertedType))
+            {
+                mutablePotentialConversionTypes.Add(targetNodeInfo.ConvertedType);
+            }
+            else if (diagnosticId == CS1503
+                && targetNode.GetAncestorsOrThis<ArgumentSyntax>().FirstOrDefault() is ArgumentSyntax targetArgument
+                && targetArgument.Parent is ArgumentListSyntax argumentList
+                && argumentList.Parent is SyntaxNode invocationNode) // invocation node could be Invocation Expression, Object Creation, Base Constructor...
+            {
+                // Implicit downcast appears on the argument of invocation node, get all candidate functions and extract potential conversion types 
+                var symbolInfo = semanticModel.GetSymbolInfo(invocationNode, cancellationToken);
+                var candidateSymbols = symbolInfo.CandidateSymbols;
+
+                foreach (var candidateSymbol in candidateSymbols.OfType<IMethodSymbol>())
+                {
+                    if (CanArgumentTypesBeConvertedToParameterTypesTest(
+                        semanticModel, root, argumentList, candidateSymbol.Parameters, targetArgument,
+                        cancellationToken, out var targetArgumentConversionType))
```

Comment:
```suggestion
                    if (CanArgumentTypesBeConvertedToParameterTypesTest(
                            semanticModel, root, argumentList, candidateSymbol.Parameters, targetArgument,
                            cancellationToken, out var targetArgumentConversionType))
```

Reply:
in general, child nodes that are wrapped should be indented a level.  (not a hard and fast rule, but it makes code easier o read).